addons/knowledge/models/knowledge_article.py

```diff
@@ -766,12 +768,14 @@ def invite_members(self, partner_ids, permission, send_mail=True):
         if not self.user_can_write:
             raise AccessError(_("You cannot give access to this article as you are not editor."))
         share_partner_ids = partner_ids.filtered(lambda partner: partner.partner_share)
-        article = self.sudo()
+        article = self
```

Comment:
why removing sudo here ? You need to sudo every step below and and I don't see the point.
We just tested user_can_write. So a priori, from now on, the user has complete access to the article and its member.